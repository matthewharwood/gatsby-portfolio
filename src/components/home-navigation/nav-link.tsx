import React, { FunctionComponent } from 'react';
import { useStyletron } from 'baseui';
import { Link } from 'gatsby';

type NavLinkProps = {
  to: string;
  text: string;
  index: string;
};

const NavLinkIndex: FunctionComponent = ({ children }) => {
  const [css, theme] = useStyletron();
  const c = css({
    position: 'absolute',
    zIndex: theme.zIndex.modal,
  });
  return <i className={c}>{children}</i>;
};

const NavLinkSpan: FunctionComponent = ({ children }) => {
  const [css, theme] = useStyletron();
  const c = css({
    paddingTop: 0,
    paddingRight: theme.sizing.scale1400,
    paddingBottom: 0,
    paddingLeft: 0,
    display: 'flex',
    justifyContent: 'flex-end',
    position: 'relative',
  });
  return <span className={c}>{children}</span>;
};

const NavLink: FunctionComponent<NavLinkProps> = ({
  text = '',
  index = '',
  to = '/',
  children,
}) => {
  const [css, theme] = useStyletron();
  const c = css({
    cursor: 'pointer',
    display: 'flex',
    position: 'relative',
    textDecoration: 'none',
  });
  return (
    <Link to={to} className={c}>
      <NavLinkSpan>{text}</NavLinkSpan>
      <NavLinkIndex>{index}</NavLinkIndex>
    </Link>
  );
};

export { NavLink };
