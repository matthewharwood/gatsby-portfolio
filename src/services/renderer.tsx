function componentMapFactory() {
  const componentMap = {
    "postHeader": (props: any) => console.log("POSTHEADER", props),
    "summary": (props: any) => console.log("SUMMARY", props),
  }

  let defaultValueHandler = {
    get: function(target: any, name: any) {
      return target.hasOwnProperty(name)
        ? target[name]
        : (props: any) => console.warn(
          "We are missing a component type: ", props._type, "is not found"
        );
    },
  }

  return new Proxy(componentMap, defaultValueHandler)
}

export const pageRenderer = (raw) => {

  const componentMap = componentMapFactory();

  raw.map((d) => [d._type, d]).map(([key, props]) => componentMap[key](props));
  return true
}