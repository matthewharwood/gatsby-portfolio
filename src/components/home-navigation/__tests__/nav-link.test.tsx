import React from 'react';
import renderer from 'react-test-renderer';
import { NavLink } from '../nav-link';

describe('Should create Nav with children', () => {
  it('renders correctly', () => {
    const FIXTURE = {
      text: 'Hello',
      index: '01',
      to: '/hello',
    };
    const tree = renderer.create(<NavLink {...FIXTURE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
