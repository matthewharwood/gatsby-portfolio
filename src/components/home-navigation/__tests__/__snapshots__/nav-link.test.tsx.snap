// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create Nav with children renders correctly 1`] = `
<a
  className=""
  href="/hello"
>
  <span
    className=""
  >
    Hello
  </span>
  <i
    className=""
  >
    01
  </i>
</a>
`;
